server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1744,18 +1744,41 @@ public void importMapFiles(long tid, Map<TabletFile,MapFileInfo> fileMap, boolea
 
     // Clients timeout and will think that this operation failed.
     // Don't do it if we spent too long waiting for the lock
-    long now = System.currentTimeMillis();
+    long now = System.nanoTime();
     synchronized (this) {
       if (isClosed()) {
         throw new IOException("tablet " + extent + " is closed");
       }
 
-      // TODO check seems unneeded now - ACCUMULO-1291
-      long lockWait = System.currentTimeMillis() - now;
-      if (lockWait
-          > getTabletServer().getConfiguration().getTimeInMillis(Property.GENERAL_RPC_TIMEOUT)) {
-        throw new IOException(
-            "Timeout waiting " + (lockWait / 1000.) + " seconds to get tablet lock for " + extent);
+      long rpcTimeoutNanos = TimeUnit.MILLISECONDS.toNanos(
+          (long) (getTabletServer().getConfiguration().getTimeInMillis(Property.GENERAL_RPC_TIMEOUT)
+              * 1.1));
+
+      // wait for any files that are bulk importing up to the RPC timeout limit
+      while (!Collections.disjoint(bulkImporting, fileMap.keySet())) {
+        try {
+          wait(50);
```

Comment:
Would this benefit from a retry with a back-off, clamped to 500 or 1_000 ms?   Maybe something like [50, 100, 200, 400, 500, 500, ....] up until the rpc timeout?  This could still be responsive, but also allow some breathing room when calculating the disjoint - particularly if it is a large file set?

Reply:
That's a good point. 
`GENERAL_RPC_TIMEOUT` has a default value of `120s` but it is reasonable to think that a user specified timeout would never be lower than a second, so increasing this wait to a second should be fine.